packages/@aws-cdk/aws-lambda-python/test/bundling.test.ts

```diff
@@ -140,6 +140,20 @@ describe('Dependency detection', () => {
     expect(hasDependencies('/asset-input')).toEqual(true);
   });
 
+  test('Detects poetry', () => {
+    existsSyncMock.mockImplementation((p: fs.PathLike) => {
+      if (
+        /pyproject.toml/.test(p.toString()) &&
+        /poetry.lock/.test(p.toString())
+      ) {
+        return true;
+      }
+      return existsSyncOriginal(p);
+    });
+
+    expect(hasDependencies('/asset-input')).toEqual(true);
```

Comment:
Test is failing here. Not sure why 🤔. Maybe my `if` condition in the mock is incorrect? The input path is expected to have both `pyproject.toml` and `poetry.lock` files.